### PR TITLE
koordlet: collect metric for host application memory usage with page cache

### PR DESCRIPTION
**Pull request by j4ckstraw**
### Ⅰ. Describe what this PR does

collect host application memory usage with page cache.

rely on https://github.com/koordinator-sh/koordinator/pull/2278 

### V. Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `make test`

### CODE REVIEW

**Comment by saintube** [comment]
consider memoryCalculatePolicy

**Comment by j4ckstraw** [comment]
convert to ready after https://github.com/koordinator-sh/koordinator/pull/2278 merged

**Comment by j4ckstraw** [comment]
> consider memoryCollectPolicy

@saintube PTAL, thx

File: pkg/koordlet/metricsadvisor/collectors/hostapplication/host_app_collector.go [hunk 1]
```diff
@@ -138,17 +153,34 @@ func (h *hostAppCollector) collectHostAppResUsed() {
...
 		allCPUUsageCores.Value += cpuUsageValue
-		allMemoryUsage.Value += float64(memoryUsageValue)
+		// sum memory usage according to NodeMemoryCollectPolicy
+		switch nodeMemoryCollectPolicy {
+		case slov1alpha1.UsageWithoutPageCache:
+			allMemoryUsage.Value += float64(memoryUsageValue)
+		case slov1alpha1.UsageWithPageCache:
+			allMemoryUsage.Value += float64(memUsageWithPageCache)
+		default:
+			klog.Warning("unrecognized node memory colleect policy, use UsageWithoutPageCache as default")
```
Comment (saintube) [inline pkg/koordlet/metricsadvisor/collectors/hostapplication/host_app_collector.go:181]: typo: `colleect`

Comment (j4ckstraw) [reply pkg/koordlet/metricsadvisor/collectors/hostapplication/host_app_collector.go]: fixed

**Review by saintube — APPROVED** [review]

**Pull request merged**